src/app/users/get-user.ts

```diff
@@ -1,65 +1,99 @@
-import {
-  SemanticAttributes,
-  asyncRunInSpan,
-  addAttributesToCurrentSpan,
-  ENDUSER_ALIAS,
-} from "@services/tracing"
+import { SemanticAttributes, asyncRunInSpan } from "@services/tracing"
 import { UsersRepository } from "@services/mongoose"
-import { IpFetcher } from "@services/ipfetcher"
 import { getIpConfig } from "@config/app"
+import { IpFetcher } from "@services/ipfetcher"
+import { UsersIpRepository } from "@services/mongoose/users-ip"
 import { RepositoryError } from "@domain/errors"
 
 const users = UsersRepository()
+const usersIp = UsersIpRepository()
 
 export const getUserForLogin = async ({
   userId,
   ip,
+  logger,
 }: {
-  userId: string
```

Comment:
We're passing in a string here because this is the boundary to the 'type-safe' world. Conversion of primitive types to domain types happen within the `app` layer as it may require domain specific checks that would be reported back to the outer layer.

Reply:
I'm not sure to understand. which layer is typesafe, and which one is not?

graphql is not type safe? app should not be as well? or app is where the conversion is happening?

Reply:
Conversation happens in app layer. Instructed user input gets checked / validated and if all is good converted to correct type. Internally we trust our types because they are checked as a first thing inside the app boundary.

Reply:
there is basically no check being done here at the app level previously. 

doing a `await users.findById(userId as UserId)` is basically doing the same thing?

Reply:
Correct - but still its happening at the correct layer. If we wanted to add a check we wouldn't have to investigate all the client code to make sure the check is added everywhere before calling into app.

Reply:
it's also happening at the correct layer now? the `string` is passed from graphql, and casted as `UserId` when entering this function

Reply:
AFAICT this function receives a `userId: UserId` - it should be casted _within_ this function.